dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFields.java

```diff
@@ -135,9 +138,40 @@ public SubSnpCoreFields(long subSnpId, Orientation subSnpOrientation, Long snpId
         this.hgvsTStart = hgvsTStart;
         this.hgvsTStop = hgvsTStop;
         this.hgvsTOrientation = hgvsTOrientation;
+        this.genotypeList = getGenotypeListFromString(genotypes);
         this.batch = batch;
     }
 
+    private List<Map<String, String>> getGenotypeListFromString (String genotypes_string) {
+        if (genotypes_string != null && !genotypes_string.isEmpty()) {
+            boolean alleleOrientation = getAlleleOrientation();
+            String ref = getReferenceInForwardStrand();
+            String alt = getAlternateInForwardStrand();
+            return Arrays.stream(genotypes_string.split(",", -1))
+                    .map(String::trim)
+                    .map(genotypeString -> getForwardOrientedGenotype(alleleOrientation, genotypeString))
+                    .map(genotypeString -> getGenotypeMapFromString(genotypeString, ref, alt))
+                    .collect(Collectors.toList());
+        }
+        return new ArrayList<>();
+    }
+
+    private String getForwardOrientedGenotype(boolean forward, String genotypeString) {
+        Pattern genotypePattern = Pattern.compile("/|\\|");
+        String outputDelimiterToUse = genotypeString.contains("|") ? "|" : "/";
+
+        return Arrays.stream(genotypePattern.split(genotypeString, -1))
+                .map(String::trim)
+                .map(alleles -> getForwardOrientedAllele(forward, alleles))
```

Comment:
Combining the map and collect is redundant with what `getForwardOrientedAllele` already does. We could add the delimiter to the arguments list, and even overload it with and without it. If the default was '/' we wouldn't even need to change the calls from any other method.